Tests/Engine/RealTimePriceUpdateTests.cs

```diff
@@ -0,0 +1,117 @@
+            {
+                sunday, monday, tuesday, wednesday, thursday, friday
+            }.ToDictionary(x => x.DayOfWeek), earlyCloses);
+
+            _liveTradingDataFeed = new TestableLiveTradingDataFeed();
+
+            var jobPacket = new LiveNodePacket()
+            {
+                DeployId = "",
+                Brokerage = BrokerageName.OandaBrokerage.ToString(),
+                DataQueueHandler = "LiveDataQueue"
+            };
+
+            var algo = new BenchmarkAlgorithm();
+
+            _liveTradingDataFeed.Initialize(algo, jobPacket, new LiveTradingResultHandler(), new LocalDiskMapFileProvider(), null, new DefaultDataProvider());
+
+            algo.Initialize();
+        }
+
+        [TestFixtureTearDown]
+        public void TearDown()
+        {
+            _liveTradingDataFeed.Exit();
+        }
+
+
+        [Test]
+        public void NullSubscriptions_DoNotIndicateRealTimePriceUpdates()
+        {
+            Assert.IsFalse(_liveTradingDataFeed.UpdateRealTimePrice(null, new TimeZoneOffsetProviderNeverOpen()));
+        }
+
+        [Test]
+        public void ClosedExchanges_DoNotIndicateRealTimePriceUpdates()
+        {
+            var security = new Security(Symbol.Empty, _exchangeHours, new Cash("USA", 100m, 1m), SymbolProperties.GetDefault("USA"));
+            var subscription = new Subscription(null, security, null, null, new TimeZoneOffsetProviderNeverOpen(), DateTime.MinValue, DateTime.MaxValue, false);
+            Assert.IsFalse(_liveTradingDataFeed.UpdateRealTimePrice(subscription, new TimeZoneOffsetProviderNeverOpen()));
+        }
+
+        [Test]
+        public void OpenExchanges_DoIndicateRealTimePriceUpdates()
+        {
+            var security = new Security(Symbol.Empty, _exchangeHours, new Cash("USA", 100m, 1m), SymbolProperties.GetDefault("USA"));
+            var subscription = new Subscription(null, security, null, null, new TimeZoneOffsetProviderAlwaysOpen(), DateTime.MinValue, DateTime.MaxValue, false);
+            Assert.IsTrue(_liveTradingDataFeed.UpdateRealTimePrice(subscription, new TimeZoneOffsetProviderAlwaysOpen()));
+        }
+
+        class TestableLiveTradingDataFeed : LiveTradingDataFeed
+        {
+            public bool UpdateRealTimePrice(Subscription subscription, TimeZoneOffsetProvider timeZoneOffsetProvider)
+            {
+                return SubscriptionShouldUpdateRealTimePrice(subscription, timeZoneOffsetProvider);
+            }
+        }
+
+        class TimeZoneOffsetProviderNeverOpen : TimeZoneOffsetProvider
+        {
+            public TimeZoneOffsetProviderNeverOpen() 
+                : base(TimeZones.NewYork, DateTime.Parse("1/1/2016"), DateTime.Parse("1/1/2018"))
+            {
+            }
+
+            public override DateTime ConvertFromUtc(DateTime utcTime)
+            {
+                // return a date that's always closed for equities
+                return DateTime.Parse("3/18/2017 11:00:00 PM");
```

Comment:
To avoid an exception with international users, we should specify `CultureInfo.InvariantCulture` here, or use this DateTime constructor: `return new DateTime(2017, 3, 18, 23, 0, 0);`